### ISSUE TITLE
How to set network proxy?

### ISSUE THREAD

**Issue by onism0106** [781flyingdutchman/background_downloader#99]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by 781flyingdutchman** [comment]
Can you clarify your question?

**Comment by onism0106** [comment]
For example, when I make an http request, I can set up a proxy to get web content. 

final HttpClient httpClient = HttpClient();
    httpClient.findProxy = (uri) {
      return 'PROXY 127.0.0.1:7890';
    };
    final http.Client client = IOClient(httpClient);
    http.StreamedResponse response =
        await client.send(http.Request('GET', Uri.parse(downloadUrl)));


How do I set up a proxy when downloading a file?

**Comment by 781flyingdutchman** [comment]
Unfortunately I don't know of a way to accomplish this. Depending on the platform, we'r eeither using a native Android, native iOS or a Dart http client, and neither can be easily configured 'from the outside'. If you figure out how to do this, please let me know. For now, I'm closing this issue as 'not planned' - sorry.

**Comment by 781flyingdutchman** [comment]
Hi @onism0106 , I am working in a mechanism to 'configure' the downloader, and that mechanism could be used to configure a proxy. I am happy to work on it, but since I don't have a way to test this I only want to do that if you are able to test it as I develop it, on Android, iOS and desktop.  Can you let me know if you are up for that, or not?

**Comment by onism0106** [comment]
Yes, I'd love to test it. I'm looking forward to it！

**Comment by 781flyingdutchman** [comment]
OK, ready for testing.  Please use branch `config` by changing your pubspec.yaml to:
```
background_downloader:
    git:
      url: https://github.com/781flyingdutchman/background_downloader.git
      ref: config
```

Also, please read the `CONFIG.md` on that branch to understand how to configure the downloader.  There are several options (please feel free to test them all!) but the most important one for you is to set the proxy. You can do this globally (i.e. for all platforms) by calling during your app's startup:
```
FileDownloader().configure(
        globalConfig: [
          ('proxy', ('localhost', 8080)),
          // ('proxy', false), // to turn proxy off
        ]);
```

See the example app `main.dart` for a few other options.  Let me know if this works, or if you run into issues. Thanks for your help!

**Comment by onism0106** [comment]
I think it works for me ! The platform is macOS.

**Comment by 781flyingdutchman** [comment]
Great! Do you have any way to test whether it also works on Android and
iOS? Even making a super simple app that pings something via your proxy
would help validate that it works for everyone before letting it loose 😀

**Comment by 781flyingdutchman** [comment]
Actually if you can just run the example app on Android and iOS and add your proxy configuration at the start, that would test it. If I put in localhost as a proxy then the download fails, so it seems to work, but I cannot test it with a real proxy (when the download should succeed). Thanks!

**Comment by onism0106** [comment]
I'm so sorry, but for some reason I can only test on macOS. I'm sorry I can't help you.

**Comment by 781flyingdutchman** [comment]
Ok, just to clarify: do you mean the proxy does not work on Android and iOS when you test it, or do you mean that you cannot test on Android and iOS at all?

**Comment by onism0106** [comment]
I mean I cannot test on Android and iOS Platforms.

**Comment by 781flyingdutchman** [comment]
Included in V7.9.0, untested on iOS and Desktop
